client/client_pool/src/external_client.cpp

```diff
@@ -149,17 +154,21 @@ std::pair<int8_t, ConcordClient::PendingReplies> ConcordClient::SendPendingReque
   }
   try {
     auto res = new_client_->sendBatch(request_queue, batch_cid);
-    for (auto rep : res) {
-      for (auto reply : pending_replies_) {
-        if (reply.lengthOfReplyBuffer == max_reply_size_) continue;
```

Comment:
This line was not intended to determine if the returned reply exceeded the limit, but to determine if we even want to do anything with it.
The agreement with DA was to use some internal buffer (like a garbage can) to save the response when the parameter reply_size was set to zero.
Here, the reply size is set to the max_size based on the idea that DA should only send buffer with less than the max size.

Reply:
`if (reply_size) {
    pending_reply.replyBuffer = reply_buffer;
    pending_reply.lengthOfReplyBuffer = reply_size;
  } else {
    pending_reply.replyBuffer = reply_->data() + batching_buffer_reply_offset_ * max_reply_size_;
    LOG_DEBUG(logger_,
              "Given reply size is 0, setting internal buffer with offset="
                  << (batching_buffer_reply_offset_ * max_reply_size_));
    ++batching_buffer_reply_offset_;
    pending_reply.lengthOfReplyBuffer = max_reply_size_;
  }`

This is the part of the code that we do that.

Reply:
You are right, Lior. But a more correct check would be for the existence of a callback.

Reply:
Thanks!